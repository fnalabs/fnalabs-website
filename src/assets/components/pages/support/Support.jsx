import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Helmet from 'react-helmet'
import ReactGA from 'react-ga'

import Consent from 'contexts/Consent'

import { SupportHero } from './SupportHero'
import { SupportMenu } from './SupportMenu'

import meta from 'metadata'

export default class Projects extends Component {
  static contextType = Consent
  static contextTypes = {
    isConsent: PropTypes.bool
  }

  componentDidMount () {
    if (this.context.isConsent) {
      const title = `${meta.common.siteName} | ${meta['/support'].title}`
      ReactGA.pageview(this.props.location.pathname, undefined, title)
    }
  }

  render () {
    const { title, description, url } = meta['/support']
    const siteName = meta.common.siteName

    return (
      <>
        <Helmet>
          <title>{siteName} | {title}</title>
          <meta name='description' content={description} />
          <link rel='canonical' href={url} />

          <meta property='og:title' content={`${siteName} | ${title}`} />
          <meta property='og:description' content={description} />
          <meta property='og:site_name' content={siteName} />
          <meta property='og:url' content={url} />
          <meta property='og:type' content='website' />
        </Helmet>

        <SupportHero />
        <SupportMenu />

      </>
    )
  }
}
