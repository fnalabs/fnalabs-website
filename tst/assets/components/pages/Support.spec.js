/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

import { Cookies } from 'react-cookie-consent'
import ReactGA from 'react-ga'

import Support from '../../../../src/assets/components/pages/support/Support.jsx'

jest.mock('react-ga')

test('<Support /> - should render Support page static content', () => {
  Cookies.get = jest.fn().mockReturnValue(false)

  const tree = renderer.create(<Support location={{ pathname: '/support' }} />).toJSON()

  expect(tree).toMatchSnapshot()
  expect(ReactGA.pageview).not.toBeCalled()
})

test('<Support /> - should render Support page static content and report page view', () => {
  Cookies.get = jest.fn().mockReturnValue(true)

  const tree = renderer.create(<Support location={{ pathname: '/support' }} />).toJSON()

  expect(tree).toMatchSnapshot()
  expect(ReactGA.pageview).toBeCalledTimes(1)
})
