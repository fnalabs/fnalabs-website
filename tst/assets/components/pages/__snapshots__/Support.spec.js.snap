// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Support /> should render Support page static content 1`] = `
Array [
  <section
    className="hero is-light is-bold has-nav-spacing"
  >
    <div
      className="hero-body"
    >
      <div
        className="container"
      >
        <div
          className="columns"
        >
          <div
            className="column is-8-desktop is-offset-2-desktop"
          >
            <div
              className="tile is-ancestor"
            >
              <div
                className="tile is-parent is-6"
              >
                <div
                  className="tile is-child has-text-centered-mobile has-text-right"
                >
                  <svg
                    className="svg-hero"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M256 8C119.033 8 8 119.033 8 256s111.033 248 248 248 248-111.033 248-248S392.967 8 256 8zm173.696 119.559l-63.399 63.399c-10.987-18.559-26.67-34.252-45.255-45.255l63.399-63.399a218.396 218.396 0 0 1 45.255 45.255zM256 352c-53.019 0-96-42.981-96-96s42.981-96 96-96 96 42.981 96 96-42.981 96-96 96zM127.559 82.304l63.399 63.399c-18.559 10.987-34.252 26.67-45.255 45.255l-63.399-63.399a218.372 218.372 0 0 1 45.255-45.255zM82.304 384.441l63.399-63.399c10.987 18.559 26.67 34.252 45.255 45.255l-63.399 63.399a218.396 218.396 0 0 1-45.255-45.255zm302.137 45.255l-63.399-63.399c18.559-10.987 34.252-26.67 45.255-45.255l63.399 63.399a218.403 218.403 0 0 1-45.255 45.255z"
                      fill="#00008b"
                    />
                  </svg>
                </div>
              </div>
              <div
                className="tile is-parent is-vertical"
              >
                <div
                  className="tile is-child has-text-centered-mobile"
                >
                  <article
                    className="content"
                  >
                    <h1
                      className="title"
                    >
                      Support
                    </h1>
                    <p
                      className="subtitle"
                    >
                      How can we help you?
                    </p>
                  </article>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <section
    className="section"
  >
    <div
      className="container"
    >
      <div
        className="columns"
      >
        <div
          className="column is-10-desktop is-offset-1-desktop"
        >
          <h2
            className="title"
          >
            Support Links
          </h2>
          <div
            className="columns"
          >
            <div
              className="column"
            />
            <div
              className="column is-narrow"
            >
              <h3
                className="subtitle is-4"
              >
                Featured Projects
              </h3>
              <nav
                className="menu"
              >
                <p
                  className="menu-label"
                >
                  Hive
                  <sup>
                    io
                  </sup>
                   Framework
                </p>
                <ul
                  className="menu-list"
                >
                  <li>
                    <a
                      href="https://hiveframework.io"
                    >
                      Homepage
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://hiveframework.io/overview"
                    >
                      Overview
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://hiveframework.io/start"
                    >
                      Get Started
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://hiveframework.io/documentation"
                    >
                      Documentation
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://github.com/fnalabs/hive-io"
                    >
                      Github
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://github.com/fnalabs/hive-io/issues"
                    >
                      Issues
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://www.npmjs.com/package/hive-io"
                    >
                      Class Library (NPM)
                    </a>
                  </li>
                  <li>
                    <a
                      className="menu-label-sub"
                    >
                      Docker images
                    </a>
                    <ul>
                      <li>
                        <a
                          href="https://hub.docker.com/r/fnalabs/hive-base-js/"
                        >
                          Base
                        </a>
                      </li>
                      <li>
                        <a
                          href="https://hub.docker.com/r/fnalabs/hive-producer-js/"
                        >
                          Producer
                        </a>
                      </li>
                      <li>
                        <a
                          href="https://hub.docker.com/r/fnalabs/hive-consumer-js/"
                        >
                          Consumer
                        </a>
                      </li>
                      <li>
                        <a
                          href="https://hub.docker.com/r/fnalabs/hive-stream-processor-js/"
                        >
                          Stream Processor
                        </a>
                      </li>
                    </ul>
                  </li>
                </ul>
                <p
                  className="menu-label"
                >
                  Model(JSON)
                  <sup>
                    js
                  </sup>
                </p>
                <ul
                  className="menu-list"
                >
                  <li>
                    <a
                      href="https://fnalabs.github.io/model-json-js/"
                    >
                      Documentation
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://www.npmjs.com/package/model-json-js"
                    >
                      NPM
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://github.com/fnalabs/lib-js/tree/master/packages/model-json-js"
                    >
                      Github
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://github.com/fnalabs/lib-js/issues"
                    >
                      Issues
                    </a>
                  </li>
                </ul>
                <p
                  className="menu-label"
                >
                  Schema(JSON)
                  <sup>
                    js
                  </sup>
                </p>
                <ul
                  className="menu-list"
                >
                  <li>
                    <a
                      href="https://fnalabs.github.io/schema-json-js/"
                    >
                      Documentation
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://www.npmjs.com/package/schema-json-js"
                    >
                      NPM
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://github.com/fnalabs/lib-js/tree/master/packages/schema-json-js"
                    >
                      Github
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://github.com/fnalabs/lib-js/issues"
                    >
                      Issues
                    </a>
                  </li>
                </ul>
              </nav>
            </div>
            <div
              className="column"
            />
            <div
              className="column is-narrow"
            >
              <h3
                className="subtitle is-4"
              >
                More Projects
              </h3>
              <nav
                className="menu"
              >
                <p
                  className="menu-label"
                >
                  test-automation
                </p>
                <ul
                  className="menu-list"
                >
                  <li>
                    <a
                      href="https://fnalabs.github.io/test-automation/"
                    >
                      Documentation
                    </a>
                  </li>
                  <li>
                    <a
                      className="menu-label-sub"
                    >
                      Class Library
                    </a>
                    <ul>
                      <li>
                        <a
                          href="https://www.npmjs.com/package/test-automation"
                        >
                          NPM
                        </a>
                      </li>
                      <li>
                        <a
                          href="https://github.com/fnalabs/test-automation"
                        >
                          Github
                        </a>
                      </li>
                      <li>
                        <a
                          href="https://github.com/fnalabs/test-automation/issues"
                        >
                          Issues
                        </a>
                      </li>
                    </ul>
                  </li>
                  <li>
                    <a
                      className="menu-label-sub"
                    >
                      Test image
                    </a>
                    <ul>
                      <li>
                        <a
                          href="https://github.com/fnalabs/test-automation-starter"
                        >
                          Starter Kit
                        </a>
                      </li>
                      <li>
                        <a
                          href="https://github.com/fnalabs/test-automation-starter/issues"
                        >
                          Issues
                        </a>
                      </li>
                    </ul>
                  </li>
                </ul>
                <p
                  className="menu-label"
                >
                  Starter Kits
                </p>
                <ul
                  className="menu-list"
                >
                  <li>
                    <a
                      href="https://github.com/fnalabs/starter-npm-js"
                    >
                      NPM package
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://github.com/fnalabs/starter-react-js"
                    >
                      React SSR PWA
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://github.com/fnalabs/starter-fastify-js"
                    >
                      Fastify service
                    </a>
                  </li>
                  <li>
                    <a
                      href="https://github.com/fnalabs/starter-koa2-js"
                    >
                      Koa2 service
                    </a>
                  </li>
                </ul>
              </nav>
              <h3
                className="subtitle is-4 has-nav-spacing"
              >
                Contact By Email
              </h3>
              <nav
                className="menu"
              >
                <ul
                  className="menu-list"
                >
                  <li>
                    <a
                      href="mailto:support@fnalabs.com"
                    >
                      Support
                    </a>
                  </li>
                  <li>
                    <a
                      href="mailto:contact@fnalabs.com"
                    >
                      General
                    </a>
                  </li>
                </ul>
              </nav>
            </div>
            <div
              className="column"
            />
          </div>
        </div>
      </div>
    </div>
  </section>,
]
`;
